js_modules/dagit/packages/core/src/runs/RunsFilterInputNew.tsx

```diff
@@ -253,36 +253,125 @@ export const RunsFilterInput: React.FC<RunsFilterInputProps> = ({
     };
   }, [options]);
 
+  const jobFilter = useStaticSetFilter({
```

Comment:
Now that the state is preserved I need to remove the conditional hook usage. The hooks are pretty lightweight, all the work happens when they're actually used so its not big overhead. I should probably not fetch values for filters that aren't being used though